File: shared/domain/audit.go
package domain

import "time"

// AuditAction は監査対象の操作種別。
type AuditAction string

const (
	AuditActionRoleChange   AuditAction = "role_change"
	AuditActionAreaEdit     AuditAction = "area_edit"
	AuditActionApproval     AuditAction = "approval"
	AuditActionDoNotVisit   AuditAction = "do_not_visit"
	AuditActionForceReturn  AuditAction = "force_return"
)

// AuditLog は重要操作の履歴。
type AuditLog struct {
	ID        string      `json:"id"`
	Action    AuditAction `json:"action"`
	ActorID   string      `json:"actorId"`  // 操作者
	TargetID  string      `json:"targetId"` // 操作対象
	Detail    string      `json:"detail"`
	CreatedAt time.Time   `json:"createdAt"`
}

File: pwa/src/domain/models/audit.ts
// 監査ログのドメインモデル。
// 仕様: docs/wants/07_通知と申請.md
// 参照実装: shared/domain/models/audit.go

/** 監査対象の操作種別。 */
export type AuditAction =
  | "role_change"
  | "area_edit"
  | "approval"
  | "do_not_visit"
  | "force_return";

/** 重要操作の履歴。 */
export interface AuditLog {
  id: string;
  /** 所属領域（GroupShareのtopic） */
  regionId: string;
  action: AuditAction;
  /** 操作者 */
  actorId: string;
  /** 操作対象 */
  targetId: string;
  detail: string;
  /** ISO 8601 */
  timestamp: string;
  /** ISO 8601 */
  createdAt: string;
}
